### PR TITLE
Wait for at least 1 endpoint in E2E test for examples

### LINKED ISSUE #14910: Spark E2E test is flaky
Workers occassionaly have problems with connecting to the master
```
Oct  1 17:34:50.030: INFO: starting org.apache.spark.deploy.worker.Worker, logging to /opt/spark-1.4.0-bin-hadoop2.6/sbin/../logs/spark--org.apache.spark.deploy.worker.Worker-1-spark-worker-controller-znvxf.out
15/10/01 15:26:45 INFO SecurityManager: Changing view acls to: root
15/10/01 15:26:45 INFO SecurityManager: Changing modify acls to: root
15/10/01 15:26:45 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: Set(root); users with modify permissions: Set(root)
15/10/01 15:26:46 INFO Slf4jLogger: Slf4jLogger started
15/10/01 15:26:46 INFO Remoting: Starting remoting
15/10/01 15:26:46 INFO Remoting: Remoting started; listening on addresses :[akka.tcp://sparkWorker@10.245.0.13:42267]
15/10/01 15:26:46 INFO Utils: Successfully started service 'sparkWorker' on port 42267.
15/10/01 15:26:46 INFO Worker: Starting Spark worker 10.245.0.13:42267 with 2 cores, 6.3 GB RAM
15/10/01 15:26:46 INFO Worker: Running Spark version 1.4.0
15/10/01 15:26:46 INFO Worker: Spark home: /opt/spark-1.4.0-bin-hadoop2.6
15/10/01 15:26:46 INFO Utils: Successfully started service 'WorkerUI' on port 8081.
15/10/01 15:26:46 INFO WorkerWebUI: Started WorkerWebUI at http://10.245.0.13:8081
15/10/01 15:26:46 INFO Worker: Connecting to master akka.tcp://sparkMaster@spark-master:7077/user/Master...
15/10/01 15:26:56 INFO Worker: Retrying connection to master (attempt # 1)
15/10/01 15:26:56 INFO Worker: Connecting to master akka.tcp://sparkMaster@spark-master:7077/user/Master...
15/10/01 15:27:06 INFO Worker: Retrying connection to master (attempt # 2)
15/10/01 15:27:06 INFO Worker: Connecting to master akka.tcp://sparkMaster@spark-master:7077/user/Master...
15/10/01 15:27:16 INFO Worker: Retrying connection to master (attempt # 3)
15/10/01 15:27:16 INFO Worker: Connecting to master akka.tcp://sparkMaster@spark-master:7077/user/Master...
15/10/01 15:27:26 INFO Worker: Retrying connection to master (attempt # 4)
15/10/01 15:27:26 INFO Worker: Connecting to master akka.tcp://sparkMaster@spark-master:7077/user/Master...
15/10/01 15:27:36 INFO Worker: Retrying connection to master (attempt # 5)
15/10/01 15:27:36 INFO Worker: Connecting to master akka.tcp://sparkMaster@spark-master:7077/user/Master...
15/10/01 15:27:46 INFO Worker: Retrying connection to master (attempt # 6)
15/10/01 15:27:46 INFO Worker: Connecting to master akka.tcp://sparkMaster@spark-master:7077/user/Master...
15/10/01 15:28:47 INFO Worker: Retrying connection to master (attempt # 7)
15/10/01 15:28:47 INFO Worker: Connecting to master akka.tcp://sparkMaster@spark-master:7077/user/Master...
15/10/01 15:29:48 INFO Worker: Retrying connection to master (attempt # 8)
15/10/01 15:29:48 INFO Worker: Connecting to master akka.tcp://sparkMaster@spark-master:7077/user/Master...
15/10/01 15:30:49 INFO Worker: Retrying connection to master (attempt # 9)
15/10/01 15:30:49 INFO Worker: Connecting to master akka.tcp://sparkMaster@spark-master:7077/user/Master...
15/10/01 15:31:50 INFO Worker: Retrying connection to master (attempt # 10)
15/10/01 15:31:50 INFO Worker: Connecting to master akka.tcp://sparkMaster@spark-master:7077/user/Master...
15/10/01 15:32:51 INFO Worker: Retrying connection to master (attempt # 11)
15/10/01 15:32:51 INFO Worker: Connecting to master akka.tcp://sparkMaster@spark-master:7077/user/Master...
15/10/01 15:33:52 INFO Worker: Retrying connection to master (attempt # 12)
15/10/01 15:33:52 INFO Worker: Connecting to master akka.tcp://sparkMaster@spark-master:7077/user/Master...
```

### PR DESCRIPTION
**Pull request by mwielgus**
Fixes #14910

### CODE REVIEW

File: test/e2e/examples.go [hunk 1]
```diff
@@ -561,6 +574,20 @@ func prepareResourceWithReplacedString(inputFile, old, new string) string {
 	return podYaml
 }
 
+func waitForEndpoint(c *client.Client, ns, name string) error {
```
Comment (piosz) [inline test/e2e/examples.go]: How about moving this to util.go? It would be useful for autoscaling tests.

Comment (mwielgus) [inline test/e2e/examples.go]: Done

**Comment by piosz** [comment]
LGTM

File: test/e2e/util.go [hunk 2]
```diff
@@ -714,6 +715,20 @@ func waitForReplicationController(c *client.Client, namespace, name string, exis
...
+		endpoint, err := c.Endpoints(ns).Get(name)
+		Expect(err).NotTo(HaveOccurred())
+		if len(endpoint.Subsets) == 0 || len(endpoint.Subsets[0].Addresses) == 0 {
+			Logf("Endpoint %s/%s is not ready yet", ns, name)
+			continue
+		} else {
+			return nil
+		}
+	}
+	return fmt.Errorf("Failed to get entpoints for %s/%s", ns, name)
```
Comment (ixdy) [inline test/e2e/util.go]: you never check whether an error occurred or not. so if the endpoint fails to startup, the error is lost and you continue anyway.

(also nit: misspelled "endpoint")

**Pull request merged**